### PR TITLE
fix(quickadapter): isolate holdout from continual state

### LINKED ISSUE #132: quickadapter: isolate holdout evaluation from continual-learning state
## Context

QuickAdapter now evaluates a chronological outer holdout before refitting the deployed model.

## Problem

When `freqai.continual_learning=true`, `get_init_model()` supplies the final model from the preceding training window to HPO trials and the selection fit. Because FreqAI training windows overlap, that model has already trained on part of the current holdout.

With the template values (21 training days, 2-day backtest step, `test_size=0.333`), roughly five of the seven holdout days overlap the preceding model state. `holdout_rmse` is therefore not independent and can be optimistically biased.

Freqtrade documents continual learning as experimental and disabled by default.

## Proposal

Keep the default path unchanged. When an independent holdout is active, reject `continual_learning=true` with a precise configuration error, or explicitly disable independent-holdout claims for that mode.

## Acceptance

- `continual_learning=false` remains unchanged.
- The incompatible configuration cannot silently publish an "independent" `holdout_rmse`.
- Validation is performed at the configuration boundary and documented.

### PR DESCRIPTION
**Pull request by jerome-benoit**
## Summary

- cold-start HPO trials and the pre-refit selection model whenever `test_size` enables two-stage selection
- preserve FreqAI continual learning for the final deployed refit and the following economic evaluation interval
- give the changed HPO objective a stable semantic identity so legacy warm-state trials and best-parameter artifacts are ineligible
- make the canonical `continual_learning=false` setting explicit and document supported regressors and split semantics

## Root cause and exact contamination proof

FreqAI 2026.6 passes each successful live/dry-run training result to `DataDrawer.save_data()`, which stores that returned model in `model_dictionary`; the next training calls `get_init_model()` and receives it when `continual_learning=true`. Backtesting follows the same handoff when it saves models. Although the configuration schema metadata says `save_backtest_models=false`, the configuration loader leaves an omitted key absent and `IFreqaiModel` uses a runtime fallback of `true`. Instantiating QuickAdapter with the canonical template under Freqtrade 2026.6 therefore confirms that its backtest path retains the returned models.

The canonical template uses a 21-day window, a 2-day retraining step, a 5-minute timeframe, and `test_size=0.333`:

- 21 days contain 6,048 candles
- sklearn reserves `ceil(6,048 * 0.333) = 2,014` candles for the outer holdout
- the next window advances by 576 candles
- the next outer-holdout interval overlaps the preceding 21-day training window by `2,014 - 576 = 1,438` candles
- that raw interval overlap is 7,190 minutes, or 4 days 23 hours 50 minutes

Causal label-availability guards make the exact subset fitted by the preceding deployed model data-dependent; `1,438` is the raw temporal overlap, not an unconditional fitted-row count. A deterministic canonical-label oracle using QuickAdapter's real Zigzag availability metadata, smoothing lookahead and causal refit/holdout guards found 1,388 current scorable holdout rows that the preceding deployed model had already fitted (4 days 19 hours 40 minutes).

QuickAdapter passed that preceding state to every HPO trial and to the pre-refit selection fit. The outer holdout labels were not explicitly supplied to those fits, but the causally available overlap was encoded in the initial model, so `holdout_rmse` was not an independent diagnostic.

## State machine and chosen semantics

With `test_size != 0` (or `None`, which resolves to the existing `0.1` validation size):

1. HPO trials and the pre-refit candidate start from no initial model and use the inner train/validation split.
2. The outer holdout is predicted once. Both labels and predictions are inverse-transformed through the fitted label pipeline, so weighted `holdout_rmse` remains on the original label scale.
3. A fresh set of feature and label pipelines is fitted on all causally available current-window refit rows.
4. Only the final XGBoost or LightGBM refit receives the preceding deployed model supplied by FreqAI.
5. FreqAI stores and deploys only that returned final model for the following economic evaluation interval.

Candidate initialization is tied to the configured two-stage path, not to the post-purge holdout row count. If causal purging empties the outer holdout, selection remains cold and `holdout_rmse=inf`; this keeps the HPO objective semantics stable across retraining windows. With `test_size=0`, the existing single-fit continual-learning path remains unchanged. First training and `continual_learning=false` naturally have no deployment initial model. Regressors other than XGBoost and LightGBM continue to train without a prior model state.

Refit iteration accounting is derived from the cold candidate rather than from the unrelated preceding booster. If HPO, candidate fitting, or final refitting raises, FreqAI never calls `save_data()` with a replacement, so the previously deployed DataDrawer state remains intact.

## Persisted Optuna state

The changed HPO objective carries a stable semantic identity `candidate-cold-start-v1`. This identity is **not** encoded in the study name (`{identifier}-{pair}-{namespace}`, unchanged); it is recorded as the Optuna study `user_attr` `objective_identity` (written idempotently on study creation), and persisted HP best parameters are wrapped in an `{objective_identity, params}` envelope carrying the same value. In live/dry-run resume (`continuous=false`), a persisted HP study whose `objective_identity` user_attr is missing or mismatched is reset (deleted and recreated), and a best-parameter file without the matching identity is rejected; either case forces a cold start for that objective. Current-identity studies still resume normally, and failed trials do not publish parameters.

Migration (one-time, first live/dry-run after upgrade from a pre-identity `main`): a legacy bare best-parameter file (no `objective_identity`) is rejected once and a persisted HP study lacking the `objective_identity` user_attr is reset once, each forcing a single cold start per pair. No manual cleanup is required; steady state resumes on the next retrain. Backtest/hyperopt runs are unaffected (they force `continuous`).

#147 (already merged into the base branch) owns the `label` study keep/reset/error policy and its `reset_label_study_on_schema_mismatch` tunable, which this work reuses for the `label` namespace without adding or duplicating any schema/search-space mismatch policy. The `hp` objective identity is an orthogonal mechanism (always reset on identity mismatch, no new tunable).

`holdout_rmse` remains a model-selection diagnostic, not an economic result and not the error of the final deployed model. No `hp_rmse`/`holdout_rmse` semantic migration is introduced.

## Validation

- 47 focused assertions kept outside the repository, covering:
  - the exact 1,438-candle raw temporal overlap and a 1,388-row causal contamination oracle using FreqAI's real timerange semantics and QuickAdapter's Zigzag availability guards
  - first and successive fits, continual learning on/off, non-empty and causally emptied holdouts
  - cold HPO/candidate state versus continual final refit state
  - original-label-scale weighted holdout RMSE
  - legacy versus current Optuna study and best-parameter identities, persistence, and HPO failure
  - candidate/refit exception paths and unchanged deployed state
  - real XGBoost and LightGBM multi-stage refits, including proof that the preceding model object is not mutated
  - real HistGradientBoosting, NGBoost and CatBoost fits confirming that their paths ignore prior model state
- real Freqtrade 2026.6 QuickAdapter lifecycle and exact runtime dependency versions
- clean composition on current `upstream/main` `0d060f7c` (which already includes #147): 47/47 assertions passing
- `ruff check quickadapter`
- `ruff format --check quickadapter`
- `PYTHONPYCACHEPREFIX=/tmp/quickadapter-156-audit/pycache python -m compileall -q quickadapter ReforceXY`
- `git diff --check upstream/main...HEAD`

Fixes #132

### CODE REVIEW

File: quickadapter/user_data/freqaimodels/QuickAdapterRegressorV3.py [hunk 1]
```diff
@@ -2802,7 +2802,9 @@ def fit(
         validation_size = self._get_validation_size()
 
         model_training_parameters = copy.deepcopy(self.model_training_parameters)
-        init_model = self.get_init_model(dk.pair)
+        deployment_init_model = self.get_init_model(dk.pair)
+        independent_holdout = X_test is not None and not X_test.empty
+        selection_init_model = None if independent_holdout else deployment_init_model
```
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply quickadapter/user_data/freqaimodels/QuickAdapterRegressorV3.py]: Not applied as a repository test change. The canonical task explicitly requires all local tests to remain outside the repository and never be committed. The external suite now has 17 focused regressions covering first/subsequent training, continual learning on/off, non-empty and causally emptied holdouts, HPO/candidate/final-refit init values, failures, and real XGBoost/LightGBM refits. Also, a purged-empty holdout no longer takes a different initialization path: whenever `test_size` enables two-stage selection, the candidate stays cold so the objective semantics remain stable. Resolving as intentionally non-applicable under the mandated test scope.

File: README.md [hunk 2]
```diff
@@ -59,11 +59,12 @@ docker compose up -d --build
...
+| freqai.continual_learning                                      | false                         | bool                                                                                                                                                                                                         | Start each deployed refit from the preceding model state. When an outer holdout is active, HPO and the pre-refit selection model always start from a fresh state so `holdout_rmse` remains independent; only the final model refit continues from the preceding deployed state. The following economic evaluation interval therefore retains continual learning.                                                                                                                                                                                                                                            |
```
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply README.md]: Fixed in `bee36903`. The README now limits deployed-state continuation to XGBoost and LightGBM and explicitly states that HistGradientBoostingRegressor, NGBoost, and CatBoost train without a prior model state.

File: quickadapter/user_data/freqaimodels/QuickAdapterRegressorV3.py [hunk 3]
```diff
@@ -2824,7 +2826,7 @@ def fit(
                     self._optuna_config["space_reduction"],
                     self._optuna_config["space_fraction"],
                     model_path=dk.data_path,
-                    init_model=init_model,
+                    init_model=selection_init_model,
```
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply quickadapter/user_data/freqaimodels/QuickAdapterRegressorV3.py:2963]: Fixed in `bee36903`. HP study names now include the semantic identity `candidate-cold-start-v1`, and persisted HP best-parameter payloads must carry the same identity. Legacy warm-state studies/trials and raw best-parameter files remain intact but cannot be loaded, enqueued, sampled from, pruned against, or selected by the new objective. Current-identity studies still resume with `continuous=false`. This adds no mismatch policy or tunable, so #147 remains independent and mergeable in either order.

File: README.md [hunk 4]
```diff
@@ -58,11 +58,12 @@ docker compose up -d --build
...
+| freqai.data_split_parameters.test_size                         | 0.1                           | float [0,1) \| int >= 0 \| None                                                                                                                                                                              | Outer holdout size; `0` disables the holdout (single-stage fit). The same parameter reserves the chronological tail of the remaining training rows as inner validation for HPO and early stopping; a fractional value is relative to those remaining rows, not the original window. The holdout is predicted once and reported as weighted `holdout_rmse` in the original label scale; it measures the cold-started pre-refit selection model, not the refitted deployed model. `None` (sklearn dynamic sizing) applies only to `timeseries_split`; `train_test_split` requires a float or int; inner validation then falls back to `0.1`.                                                |
```
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply README.md]: Fixed in `ca634b97` (docs-only). The `test_size` row now qualifies that `0` (single-stage, no outer holdout) applies to `train_test_split` only; `timeseries_split` rejects it — its validation in `_make_timeseries_split_datasets` requires a float in `(0, 1)`, an int `>= 1`, or `None`. The displayed range stays a union qualified by the description (same pattern already used for `None`, which is `timeseries_split`-only), so no single-stage handling is implied for `timeseries_split`.
